cmd/dm-syncer/config.go

```diff
@@ -302,14 +308,14 @@ func loadMetaFile(metaFile string) (*config.Meta, error) {
 	return meta, nil
 }
 
-func (oc *oldConfig) convertToNewFormat() (*config.SubTaskConfig, error) {
+func (oc *syncerConfig) convertToNewFormat() (*config.SubTaskConfig, error) {
 	meta, err := loadMetaFile(oc.Meta)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	newTask := &config.SubTaskConfig{
-		Name:     "dm-syncer",
-		SourceID: "dm-syncer-from-old-config",
+		Name:     oc.Name,
+		SourceID: fmt.Sprintf("%s:%d", oc.From.Host, oc.From.Port),
```

Comment:
If the user config `From` to another instance in the same replica group and enabled GTID mode (supported later), then the checkpoint can't be used anymore?

Reply:
update in 41f8c7a